rumqttc/src/state.rs

```diff
@@ -28,7 +29,7 @@ pub enum StateError {
     #[error("A Subscribe packet must contain atleast one filter")]
     EmptySubscription,
     #[error("Mqtt serialization/deserialization error: {0}")]
-    Deserialization(#[from] mqttbytes::Error),
+    Deserialization(#[from] Error),
```

Comment:
same as above

Reply:
We have placed things this way to make it clear where that particular error is trickling down from.